blockchain/blockdao.go

```diff
@@ -434,9 +447,20 @@ func (dao *blockDAO) putBlock(blk *block.Block) error {
 			return errors.Wrapf(err, "error when compressing a block footer")
 		}
 	}
-	batch.Put(blockHeaderNS, hash[:], serHeader, "failed to put block header")
-	batch.Put(blockBodyNS, hash[:], serBody, "failed to put block body")
-	batch.Put(blockFooterNS, hash[:], serFooter, "failed to put block footer")
+	batchForBlock.Put(blockHeaderNS, hash[:], serHeader, "failed to put block header")
+	batchForBlock.Put(blockBodyNS, hash[:], serBody, "failed to put block body")
+	batchForBlock.Put(blockFooterNS, hash[:], serFooter, "failed to put block footer")
+	whichDB := getDBIndex(blk.Height(), dao.cfg.SplitDBLength)
+	kv, err := dao.getDBFromIndex(whichDB)
+	if err != nil {
+		return err
+	}
+	err = kv.Commit(batchForBlock)
+	if err != nil {
+		return err
+	}
+	blockHashDBKey := append(blockHashDBPrefix, hash[:]...)
```

Comment:
When lookup by blockHash, can we just lookup for height first? We will know which db to reach as long as the height is known.

Reply:
see,I'll try delete this key